File: server/src/postings/dto/return-posting.dto.ts
import { ApiProperty } from '@nestjs/swagger';
import { Exclude, Expose } from 'class-transformer';

@Exclude()
export class ReturnPostingDto {
  @ApiProperty()
  @Expose()
  userId: number;

  @ApiProperty()
  @Expose()
  text: string;

  @ApiProperty()
  @Expose()
  imgPath: string;

  @ApiProperty()
  @Expose()
  category: string;

  @ApiProperty({ type: () => Comment, isArray: true, nullable: true })
  @Expose()
  comments: Comment[];

  @ApiProperty()
  @Expose()
  createdAt: Date;
}

File: client/eyesee/app/src/main/java/com/yblee2001/eyesee/dto/posting/ReturnPostingDto.kt
package com.yblee2001.eyesee.dto.posting

import com.yblee2001.eyesee.dto.comment.ReturnCommentDto

data class ReturnPostingDto(
    val id: Int,
    val userId: Int,
    val text: String,
    val imgPath: String?,
    val category: String,
    val comments: List<ReturnCommentDto>?,
    val createdAt: String
)